data/routesPosts.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+
+const PostDB = require('./helpers/postDb.js');
+
+const routesPosts = express.Router();
+
+
+// Server.js handles the /api path. Router.js handles the posts path
+
+
+// Get all the posts
+routesPosts.get('/', async (req, res) => {
+    try {
+        const posts = await PostDB.get(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        res.status(500).json({ error: "The posts could not be retrieved." });
+    }
+});
+
+
+// Get single post by id
+routesPosts.get('/:id', async (req, res) => {
+    try {
+        const postById = await PostDB.getById(req.params.id);
+        if (postById.length === 0) {  
+            res.status(404).json({ message: "The post with the specified ID does not exist." });
+        } else {
+            res.status(200).json(postById);
+        }
+    } 
+    catch (error) { 
+        res.status(500).json({ error: "The post could not be retrieved." });
+    }
+});
+
+
+// Add a new post as user by id
+routesPosts.post('/', async (req, res) => {
+    if (!req.body.text) {
+        return res.status(409).json({ message: "Post cannot be blank!" });
+    }
+    if (!req.body.user_id) {
+        return res.status(409).json({ message: "Please include user id." });
+    }
+    try {
+        const data = await PostDB.insert(req.body);
+        res.status(200).json(`Successfully added post as user_id: ${data.user_id}`);
+    }
+    catch (error) {
+        res.status(500).json({ error: "Error adding post!" });
+    }
+});
+
+
+// Delete a post from the db
+routesPosts.delete('/:id', async (req, res) => {
+    const checkPostExists = await PostDB.getById(req.params.id);
+    if (!checkPostExists) { 
+        return res.status(404).json({ message: "The post with ID does not exist. Cannot delete!" });
+    }
+    try {
+        const postDeleted = await PostDB.remove(req.params.id);
+        if (postDeleted > 1) {
+            res.status(200).json({ message: `${postDeleted} posts have been deleted` })
+        } else {
+            res.status(200).json({ message: `${postDeleted} post has been deleted` })
+        }
+    }
+    catch (error) {
+        res.status(500).json({ message: "Something went wrong deleting post!"})
```

Comment:
this works just fine (although I'm not sure if the database is equipped to handle a multi-delete or not, but doesn;t hurt), just be aware that if instead of a message describing how many were successfully deleted, you could return the info of the deleted post since you already grabbed it with checkPostExists up above

Reply:
Cool, I improved the messaging of this endpoint to include the deleted post as well as the little message in another key/value pair to the json object.